### PR TITLE
fix: support big csv files on langflow-less ingestion

### PR DESCRIPTION
**Pull request by lucaseduoli**
This pull request enhances the document chunking and embedding workflow to ensure that text chunks sent for embedding are properly aligned with their embeddings, particularly by splitting oversized chunks before batching. It also improves the robustness of table extraction and adds comprehensive unit tests for the new chunk-splitting logic and edge cases in document processing.

**Document chunking and embedding alignment:**

* Introduced a new function `split_chunks_by_max_tokens` in `src/utils/document_processing.py` to pre-split any chunk whose token count exceeds a model-specific limit, ensuring that each chunk aligns 1-to-1 with its embedding. This function uses the appropriate tokenizer based on the embedding model.
* Integrated `split_chunks_by_max_tokens` into the document processing pipeline in `src/models/processors.py`, so that chunks are split before batching for embeddings, preventing misalignment between chunks and their embeddings. [[1]](diffhunk://#diff-c92d64751875ea78390bdb54d800625aa6dd39817344a741d2c640987e91d0f7R13) [[2]](diffhunk://#diff-c92d64751875ea78390bdb54d800625aa6dd39817344a741d2c640987e91d0f7L334-R342)

**Robustness improvements in table extraction:**

* Updated `extract_relevant` in `src/utils/document_processing.py` to safely handle cases where table data is missing (`None`), preventing errors during extraction.

**Testing:**

* Added unit tests in `tests/unit/test_document_processing_resplit.py` for `split_chunks_by_max_tokens`, covering both under-limit and over-limit scenarios, and for handling tables with missing data in `extract_relevant`.

**Refactoring and cleanup:**

* Removed an unused variable (`texts`) in `process_document_standard` after refactoring the chunk splitting logic.

Copy of #1966 

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved embedding preparation by splitting oversized document chunks to respect token limits before batching for embeddings, keeping chunk-to-embedding alignment consistent.
  * Hardened table extraction to safely handle tables with missing data.
* **Tests**
  * Added unit tests for token-based chunk splitting (including emoji/tokenization) and for safe extraction when table data is `None`.
  * Added an integration test validating large CSV table ingestion is split and indexed correctly.

### CODE REVIEW

File: src/utils/document_processing.py [hunk 1]
```diff
@@ -174,3 +176,50 @@ def resplit_chunks_character_windows(
...
+    for chunk in chunks:
+        text = chunk.get("text", "")
+        if not text:
+            new_chunks.append(chunk)
+            continue
+
+        # Quick estimate: if char count is less than max_tokens, it's definitely under max_tokens
+        if len(text) < max_tokens:
+            new_chunks.append(chunk)
+            continue
```
[the remark this answers is not in the document]
Comment (edwinjosechittilappilly) [reply src/utils/document_processing.py]: is it the same chunking stratergy that was used by langflow pipeline? can we make it consistent?

**Review by edwinjosechittilappilly — CHANGES_REQUESTED** [review]
NIT comments.

**Comment by lucaseduoli** [comment]
> is it the same chunking stratergy that was used by langflow pipeline? can we make it consistent?

No, they serve different roles:

1. **This function (`split_chunks_by_max_tokens`)** is not the primary chunking strategy. It is a **safety guardrail / token-limit safeguard** used in the Python-only ingestion pipeline. It uses `tiktoken` to ensure that no individual chunk exceeds the embedding model's context window limit (e.g., 2000 for Ollama, 8000 for OpenAI/LiteLLM models) before we call the embedding API. This prevents API truncation and guarantees that the generated embeddings map 1-to-1 to the text chunks we write to OpenSearch.
2. **Langflow's chunking** (in `ingestion_flow.json`) uses LangChain's `CharacterTextSplitter` (which splits by character separator like `\n` up to `chunkSize`).

**Regarding Consistency:**
- Because this function acts as a final token-limit safeguard, it *must* operate on token counts (via `tiktoken`) rather than character counts/separators to ensure model compatibility. 
- However, our primary chunking for plain text files (`process_text_file`) *does* split by paragraphs (`\n\n`) and merges up to `1000` characters, which mimics Langflow's default `CharacterTextSplitter` behavior. For other files, we group by pages/tables via Docling and optionally apply a sliding character window.

**Review by edwinjosechittilappilly — APPROVED** [review]
Code LGTM.

@lucaseduoli  if possible can you add integration tests for non langflow ingestion.
